src/Hint/Structure.hs

```diff
@@ -39,6 +39,7 @@ foo = let ~(x:xs) = y in z
 module Hint.Structure(structureHint) where
 
 import Hint.Type
+import Data.Function (on)
```

Comment:
I disagree. IDE's are the thing that should resolve variables, not humans, and Hoogle usually gets it most of the time. For particularly obscure names I do sometimes use explicit imports, or where I am importing some functionality which seems to be disjoint from the rest of the module. But for things like `on`, most people should probably learn what they do and where they come from.

Adding a new name is considered a non-breaking change in PVP because PVP doesn't think these things through - I consider that a bug in PVP (or the whole concept flawed...). It's still a fairly rare instance though.

And the benefit of implicit imports is that I can code without continually adjusting the import list, which is a form of administration that distracts from the real job of coding. For me, the benefit in faster coding is worth the disadvantages you list. I appreciate a lot of people disagree with me here.